api/base/serializers.py

```diff
@@ -701,6 +701,25 @@ def is_valid(self, clean_html=True, **kwargs):
 
         return ret
 
+class JSONAPIRelationshipSerializer(ser.Serializer):
+    def relationship(self, obj):
+        raise NotImplementedError
+
+    def to_representation(self, obj, envelope='data'):
+        ret = {}
+        meta = getattr(self, 'Meta', None)
+        type_ = getattr(meta, 'type_', None)
+        assert type_ is not None, 'Must define Meta.type_'
+        relationship = self.relationship(obj)
+        if relationship:
+            data = collections.OrderedDict({'type': type_, 'id': relationship._id})
+        else:
+            data = None
+        if envelope:
+            ret[envelope] = data
+        else:
+            ret = data
+        return ret
```

Comment:
L718-722 are the same as JSONAPISerializer - this is just the final touchup on the response data to see if it needs to be nested inside `data` or not.  Could this be consolidated?